### PR TITLE
Publish 4.0.0

### PR DESCRIPTION
**Pull request by daileytj**
Publishes 4.0.0

### CODE REVIEW

**Comment by jeffvg** [comment]
will we be able to retain these images for read me once we migrate?

https://raw.githubusercontent.com/pxblue/react-native-workflows/master/login-workflow/media/ios-login.png

plus six others

**Comment by daileytj** [comment]
@jeffvg I'm not sure how the Github user content will be managed when we do an org switch. I'm hopeful that these are retained and we can just swap out the org name.

**Review by jeffvg — APPROVED** [review]

**Comment by emclaug2** [comment]
@daileytj  This is the error/warning I mentioned yesterday; on a fresh clone I see this message when running `yarn start:example-android`.  Do you know what might cause this?  The example still runs, but these logs are thrown while building.  

```
Building workflow package...
warning ..\..\package.json: No license field
$ bob build
ℹ Building target commonjs
ℹ Cleaning up previous build at lib\commonjs
ℹ Compiling 62 files in src with babel
✓ Wrote files to lib\commonjs
ℹ Building target module
ℹ Cleaning up previous build at lib\module
ℹ Compiling 62 files in src with babel
✓ Wrote files to lib\module
ℹ Building target typescript
ℹ Cleaning up previous build at lib\typescript
ℹ Generating type definitions with tsc
shared-auth/src/contexts/AccountUIContext/actions.ts:44:68 - error TS2571: Object is of type 'unknown'.

44             dispatch(DispatchActions.loginFailed(email, transitId, error.message));
                                                                      ~~~~~

shared-auth/src/contexts/AccountUIContext/actions.ts:54:76 - error TS2571: Object is of type 'unknown'.

54             dispatch(DispatchActions.resetPasswordFailed(email, transitId, error.message));
                                                                              ~~~~~

shared-auth/src/contexts/AccountUIContext/actions.ts:70:71 - error TS2571: Object is of type 'unknown'.

70             dispatch(DispatchActions.verifyResetCodeFailed(transitId, error.message));
                                                                         ~~~~~

shared-auth/src/contexts/AccountUIContext/actions.ts:80:67 - error TS2571: Object is of type 'unknown'.

80             dispatch(DispatchActions.setPasswordFailed(transitId, error.message));
                                                                     ~~~~~

shared-auth/src/contexts/RegistrationUIContext/actions.ts:37:64 - error TS2571: Object is of type 'unknown'.

37             dispatch(DispatchActions.loadEulaFailed(transitId, error.message));
                                                                  ~~~~~

shared-auth/src/contexts/RegistrationUIContext/actions.ts:48:79 - error TS2571: Object is of type 'unknown'.

48             dispatch(DispatchActions.requestRegistrationCodeFailed(transitId, error.message));
                                                                                 ~~~~~

shared-auth/src/contexts/RegistrationUIContext/actions.ts:69:80 - error TS2571: Object is of type 'unknown'.

69             dispatch(DispatchActions.validateUserRegistrationFailed(transitId, error.message));
                                                                                  ~~~~~

shared-auth/src/contexts/RegistrationUIContext/actions.ts:90:68 - error TS2571: Object is of type 'unknown'.

90             dispatch(DispatchActions.registerUserFailed(transitId, error.message));
                                                                      ~~~~~

src/components/SimpleDialog.tsx:39:18 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Pick<TextProps & { children: ReactNode; } & { children: ReactNode; style?: StyleProp<TextStyle>; theme: Theme; }, "style" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; }' is missing the following properties from type 'Pick<TextProps & { children: ReactNode; } & { children: ReactNode; style?: StyleProp<TextStyle>; theme: Theme; }, "style" | ... 37 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut
  Overload 2 of 2, '(props: PropsWithChildren<Pick<TextProps & { children: ReactNode; } & { children: ReactNode; style?: StyleProp<TextStyle>; theme: Theme; }, "style" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }>, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; }' is missing the following properties from type 'Pick<TextProps & { children: ReactNode; } & { children: ReactNode; style?: StyleProp<TextStyle>; theme: Theme; }, "style" | ... 37 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut

39                 <Dialog.Title>{title}</Dialog.Title>
                    ~~~~~~~~~~~~


src/screens/ChangePassword.tsx:184:43 - error TS2571: Object is of type 'unknown'.

184             setTransitState(transitFailed(error.message));
                                              ~~~~~

src/screens/ContactSupport.tsx:127:30 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; style: { color: string; }; onPress: () => Promise<void>; }' is missing the following properties from type 'Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | "testID" | ... 36 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut
  Overload 2 of 2, '(props: PropsWithChildren<Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }>, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; style: { color: string; }; onPress: () => Promise<void>; }' is missing the following properties from type 'Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | "testID" | ... 36 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut

127                             <Text
                                 ~~~~


src/screens/ContactSupport.tsx:142:30 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; style: { color: string; }; onPress: () => Promise<void>; }' is missing the following properties from type 'Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | "testID" | ... 36 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut
  Overload 2 of 2, '(props: PropsWithChildren<Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | ... 37 more ... | "android_hyphenationFrequency"> & { ...; }>, context?: any): ReactElement<...> | ... 1 more ... | null', gave the following error.
    Type '{ children: string; style: { color: string; }; onPress: () => Promise<void>; }' is missing the following properties from type 'Pick<TextProps & { style?: StyleProp<TextStyle>; theme: Theme; } & RefAttributes<{}>, "ref" | "style" | "children" | "testID" | ... 36 more ... | "android_hyphenationFrequency">': onPressIn, onPressOut

142                             <Text
                                 ~~~~



Found 12 errors.
```

**Review by emclaug2 — APPROVED** [review]
Works on Android & ios !

File: login-workflow/CHANGELOG.md [hunk 1]
```diff
@@ -5,17 +5,29 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v4.0.0 (December 7, 2021)
```
Comment (emclaug2) [inline login-workflow/CHANGELOG.md:8]: Can we get an updated date here?

Comment (daileytj) [reply login-workflow/CHANGELOG.md]: That's what this PR is for... https://github.com/pxblue/react-native-workflows/pull/106

**Comment by jeffvg** [comment]
going to look at this again

**Review by emclaug2 — APPROVED** [review]

**Comment by emclaug2** [comment]
> going to look at this again

Sounds good, I ran it in ios & it ran without issues using the latest react-auth-shared package.

**Comment by daileytj** [comment]
Tested a bit more on my end. I'm happy with it. @jeffvg feel free to hit the button when you're satisfied.

**Comment by jeffvg** [comment]
since react auth shared is in dev @ 3.6.0 should we update the gitModules?

**Comment by daileytj** [comment]
@jeffvg hmmm... I think we can wait to do that until we get the org name change. There's a story for this. It will need updated again anyway. Feel free to try to change my mind.

**Comment by jeffvg** [comment]
> > since react auth shared is in dev @ 3.6.0 should we update the gitModules?
> 
> I don't think we need to do that for this PR, as long as the demo can run we should be okay.

ok, it works set either way @pxblue 3.5.0 or @brightlayer-ui 3.6.0

**Pull request merged**